### PR TITLE
applications: machine_learning: Add NUS to EI data forwarder

### PR DESCRIPTION
**Pull request by MarekPieta**
Change introduces forwarding sensor samples over NUS and adds basic Bluetooth configuration for Thingy52.

Jira: NCSDK-8507

### CODE REVIEW

File: applications/machine_learning/src/modules/ei_data_forwarder.c [hunk 1]
```diff
@@ -142,6 +203,109 @@ static int init(void)
 	return err;
 }
 
+void bt_nus_sent_cb(struct bt_conn *conn)
```
Comment (pdunaj) [inline applications/machine_learning/src/modules/ei_data_forwarder.c:206]: static

**Review by pdunaj — CHANGES_REQUESTED** [review]
Since NUS is so much different then UART I have a feeling it should be separated to different file. Let's talk offline.

File: applications/machine_learning/src/modules/ei_data_forwarder.c (hunk 1 above)
Comment (MarekPieta) [reply applications/machine_learning/src/modules/ei_data_forwarder.c:206]: fixed

File: applications/machine_learning/src/modules/ei_data_forwarder_bt_nus.c [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+	int pos = ei_data_forwarder_parse_data(sensor_event_get_data_ptr(event),
+					       sensor_event_get_data_cnt(event),
+					       buf,
+					       sizeof(buf));
+
+	if (pos < 0) {
+		atomic_cas(&nus_busy, true, false);
+		LOG_ERR("EI data forwader parsing error: %d", pos);
+		report_error();
+		return pos;
```
Comment (MarekPieta) [inline applications/machine_learning/src/modules/ei_data_forwarder_bt_nus.c:98]: Will change it to return false, we should not consume the event in that case

Comment (MarekPieta) [reply applications/machine_learning/src/modules/ei_data_forwarder_bt_nus.c:98]: Added fix

**Review by emob-nordic — APPROVED** [review]

File: applications/machine_learning/src/util/ei_data_forwarder.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	}
+
+	return 0;
+}
+
+int ei_data_forwarder_parse_data(const float *data_ptr, size_t data_cnt,
+				 uint8_t *buf, size_t buf_size)
+{
+	int pos = 0;
+	int tmp;
```
Comment (pdunaj) [inline applications/machine_learning/src/util/ei_data_forwarder.c:27]: tmp could be moved to lower scope

**Review by pdunaj — APPROVED** [review]

File: applications/machine_learning/src/util/ei_data_forwarder.c (hunk 3 above)
Comment (MarekPieta) [reply applications/machine_learning/src/util/ei_data_forwarder.c:27]: done

**Pull request merged**